website/project/model.py

```diff
@@ -2162,6 +2166,64 @@ def add_tag(self, tag, auth, save=True, log=True):
             if save:
                 self.save()
 
+    def add_citation(self, auth, save=True, log=True, **kwargs):
+        citation = AlternativeCitation(**kwargs)
+        citation.save()
+        self.alternative_citations.append(citation)
+        if log:
+            self.add_log(
+                action=NodeLog.CITATION_ADDED,
+                params={
+                    'node': self._primary_key,
+                    'citation': citation.name
```

Comment:
We should probably store both the name and the text of the citation, just in case we ever decide to add citation text to logs.

```python
'citation': {'name': ..., 'text': ...}
```

Same goes for the CITATION_REMOVED logs.